app/forms/nsm/steps/solicitor_declaration_form.rb

```diff
@@ -8,14 +8,27 @@ class SolicitorDeclarationForm < ::Steps::BaseFormObject
 
       def persist!
         Claim.transaction do
-          application.state = :submitted
-          application.update_work_item_positions!
-          application.update_disbursement_positions!
+          application.state = new_state
+          if application.submitted?
+            application.update_work_item_positions!
+            application.update_disbursement_positions!
+          end
           application.update!(attributes)
```

Comment:
Here's another sneaky requirement from CRM457-2004 that I think has been missed:

> And the Legal Representative is saved along with the specific RFI response as is supplemental to the original claim Legal Rep

At present this code is going to _overwrite_ the original signatory_name with the new one. As I understand it, we want to leave the original signatory_name intact and instead stick what they've entered this time round into the current further_information.